base/jobs/docker/1.0/py3/Dockerfile.cpu

```diff
@@ -1,128 +1,76 @@
- && tar -xvf Python-$PYTHON_VERSION.tgz \
- && cd Python-$PYTHON_VERSION \
- && ./configure && make && make install \
- && rm -rf ../Python-$PYTHON_VERSION*
-
-# Upgrading pip and creating symbolic link for python3
-RUN ${PIP} --no-cache-dir install --upgrade pip setuptools
-RUN ln -s $(which ${PYTHON}) /usr/local/bin/python \
- && ln -s $(which ${PIP}) /usr/bin/pip
-
-WORKDIR /
-
-RUN apt-get update && apt-get -y install cmake protobuf-compiler
+    && rm -rf /var/lib/apt/lists/*
+
+# Download and install OpenMPI
+RUN mkdir /tmp/openmpi && cd /tmp/openmpi \
+    && wget https://download.open-mpi.org/release/open-mpi/v4.0/openmpi-4.0.4.tar.gz \
+    && tar zxf openmpi-4.0.4.tar.gz && cd openmpi-4.0.4 \
+    && ./configure --enable-orterun-prefix-by-default \
+    && make -j $(nproc) all \
+    && make install \
+    && ldconfig \
+    && rm -rf /tmp/openmpi
+
+# Runtime stage
+FROM public.ecr.aws/docker/library/python:3.10
+
+# Copy installed Python and OpenMPI from the builder stage
+COPY --from=builder /usr/local /usr/local
+
+# Set environment variables
+ENV LANG=C.UTF-8 \
+    LC_ALL=C.UTF-8 \
+    PATH=/usr/local/bin:$PATH
+
+# Install runtime dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    libssl-dev \
+    ca-certificates \
+    gcc \
+    && rm -rf /var/lib/apt/lists/* \
+    && apt-get clean
 
-# Copy the dependencies file
-COPY requirements.txt /
-# Installing our custom python libraries
-RUN ${PIP} install --no-cache --upgrade \
-        -r requirements.txt
+# Prepare environment for OpenMPI and Python application
+COPY --from=builder /usr/local/bin/mpirun /usr/local/bin/mpirun
+COPY --from=builder /usr/local/etc/openmpi-mca-params.conf /usr/local/etc/openmpi-mca-params.conf
 
-        
-RUN ${PIP} install --no-cache --upgrade sagemaker-training==4.4.10
+# Set working directory and copy necessary files for the application
+WORKDIR /app
+COPY requirements.txt ./
+COPY braket_container.py /opt/ml/code/
 
-RUN HOME_DIR=/root \
- && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
- && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
- && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
- && chmod +x /usr/local/bin/testOSSCompliance \
- && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
- && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
- && rm -rf ${HOME_DIR}/oss_compliance*
+# Install Python packages
+RUN pip install --no-cache-dir -r requirements.txt \
+    && pip install --no-cache-dir sagemaker-training==4.7.4 
 
-# Setup our entry point
+# Set default command or entry point
+ENV SAGEMAKER_PROGRAM braket_container.py
 
-COPY braket_container.py /opt/ml/code/braket_container.py
+# Clean up
+RUN apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```

Comment:
Why? Any files removed will exist in previous layers so will not reduce the image size.